python/tvm/micro/session.py

```diff
@@ -143,6 +144,9 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         """Tear down this session and associated RPC session resources."""
         self.transport.__exit__(exc_type, exc_value, exc_traceback)
 
+    def _shutdown(self):
+        self.__exit__(None, None, None)
```

Comment:
perhaps for a future cleanup: let's ensure we avoid double-calling `__exit__` by checking some local var to determine whether we did already.